From ac3506536034b2a784e11f604ded145e56bb3512 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Fri, 14 Sep 2018 15:03:37 +0300
Subject: [PATCH] EEprom Layout: update to new format

Removed few not used field, others changed from bin to ascii to support
Any character and not only HEX characters (a-f)
---

diff --git a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
index 35484d8..50d17f0 100755
--- a/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
+++ b/board/habanalabs/common/rootfs_overlay/usr/sbin/eeprog
@@ -36,17 +36,13 @@
 LAYOUT_VER=`xxd -p -l1 ${LAYOUT}`
 if [ "${LAYOUT_VER}" -eq "0" ]; then
 	print_field_num "Layout Version" 0 1
-	print_field_num "Production Serial Number" 1 5
-	print_field_num "PCB Assembly Version" 6 1
-	print_field_num "CPLD FW Version" 7 1
-	print_field_num "Infineon FW Version" 8 1
-	print_field_num "DDR Type and Speed" 9 1
-	print_field_num "Fans Vendor" 10 1
-	print_field_num "Mechanics Vendor" 11 1
-	print_field_num "Thermal Solution Vendor" 12 1
-	print_field_num "PCIe Gen3 or Gen4 Enabled" 13 1
-	print_field_num "MAC Address" 14 6
-	print_field_str "Board Configuration" 20 64
+	print_field_str "Production Serial Number" 1 11
+	print_field_str "PCB Assembly Version" 12 4
+	print_field_str "PCB Version" 16 4
+	print_field_num "Infineon FW Version" 20 1
+	print_field_num "DDR Type and Speed" 21 1
+	print_field_num "PCIe Gen3 or Gen4 Enabled" 22 1
+	print_field_str "Board Configuration" 23 64
 else
 	echo "Unknown eeprom layout"
 fi
diff --git a/board/habanalabs/goya/uboot-eeprom.cmd b/board/habanalabs/goya/uboot-eeprom.cmd
index d7744f9..7f82368 100644
--- a/board/habanalabs/goya/uboot-eeprom.cmd
+++ b/board/habanalabs/goya/uboot-eeprom.cmd
@@ -2,16 +2,12 @@
 mw.b 7ff0000000 ff 100
 eeprom write  $LAYOUT 7ff0000000 0 100
 eeprom update $LAYOUT "Layout Version" "0"
-eeprom update $LAYOUT "Production Serial Number" "a011223344"
-eeprom update $LAYOUT "PCB Assembly Version" "1a"
-eeprom update $LAYOUT "CPLD FW Version" "2b"
+eeprom update $LAYOUT "Production Serial Number" "AI11223344"
+eeprom update $LAYOUT "PCB Assembly Version" "V1A"
+eeprom update $LAYOUT "PCB Version" "R0B"
 eeprom update $LAYOUT "Infineon FW Version" "3c"
 eeprom update $LAYOUT "DDR Type and Speed" "12"
-eeprom update $LAYOUT "Fans Vendor" "3"
-eeprom update $LAYOUT "Mechanics Vendor" "4"
-eeprom update $LAYOUT "Thermal Solution Vendor" "5"
-eeprom update $LAYOUT "PCIe Gen3 or Gen4 Enabled" "6"
-eeprom update $LAYOUT "MAC Address" "00:11:22:33:44:55"
+eeprom update $LAYOUT "PCIe Gen3 or Gen4 Enabled" "3"
 eeprom update $LAYOUT "Board Configuration" "Goya HL1000"
 echo "eeprom content is:"
 eeprom print  $LAYOUT
